Wikipedia/Code/SectionEditorViewController.swift

```diff
@@ -550,15 +551,21 @@ class SectionEditorViewController: ViewController {
         let alert = UIAlertController(title: CommonStrings.editorExitConfirmationTitle, message: CommonStrings.editorExitConfirmationBody, preferredStyle: .alert)
         let confirmClose = UIAlertAction(title: CommonStrings.discardEditsActionTitle, style: .destructive) { _ in
             self.closeEditor(navigateToURL: url)
+            self.logCancelEdit()
         }
         alert.addAction(confirmClose)
         let cancel = UIAlertAction(title: CommonStrings.cancelActionTitle, style: .default)
         alert.addAction(cancel)
         present(alert, animated: true)
     }
 
+    fileprivate func logCancelEdit() {
+        EditAttemptFunnel.shared.logAbort(articleURL: articleURL)
+    }
+
     fileprivate func closeEditor(navigateToURL url: URL? = nil) {
         delegate?.sectionEditorDidCancelEditing(self, navigateToURL: url)
+        logCancelEdit()
```

Comment:
Is this event needed here, as `ArticleViewController`'s implementation of `sectionEditorDidCancelEditing` itself sends the same `logAbort` event leading to duplicate calls?